### PR TITLE
fix: Access node owner by top level `owner` property

### LINKED ISSUE #47267: [Bug]: In version 30, this issue still exists.---Numerical accounts show all files/folders in "Shared" status in the web UI
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

In version 29, the issue has been resolved.
In version 30, this issue still exists.
[https://github.com/nextcloud/server/issues/45723](url)
Upgraded 29 beta 4 to 30 beta1~beta4，end of 30RC1

### Steps to reproduce

Numerical accounts show all files/folders in "Shared" status in the web UI 

### Expected behavior

NO->>Numerical accounts show all files/folders in "Shared" status in the web UI 

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

master

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.3

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 28 to 29)

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "*.*.*.*",
            "*.*.*.*"
        ],
        "maintenance_window_start": 1,
        "overwrite.cli.url": "http:\/\/*.*.*.*",
        "overwriteprotocol": "https",
        "htaccess.RewriteBase": "\/",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "30.0.0.8",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "updater.release.channel": "beta",
        "default_phone_region": "CN",
        "default_language": "zh_CN",
        "default_locale": "zh",
        "skeletondirectory": "",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379,
            "timeout": 0,
            "password": "***REMOVED SENSITIVE VALUE***"
        },
        "maintenance": false,
        "theme": "",
        "loglevel": 2
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 3.0.0
  - bruteforcesettings: 3.0.0
  - circles: 30.0.0-dev
  - cloud_federation_api: 1.13.0
  - comments: 1.20.1
  - contactsinteraction: 1.11.0
  - dav: 1.31.1
  - federatedfilesharing: 1.20.0
  - federation: 1.20.0
  - files: 2.2.0
  - files_downloadlimit: 3.0.0
  - files_pdfviewer: 3.0.0
  - files_reminders: 1.3.0
  - files_sharing: 1.22.0
  - files_trashbin: 1.20.1
  - files_versions: 1.23.0
  - logreader: 3.0.0
  - lookup_server_connector: 1.18.0
  - nextcloud_announcements: 2.0.0
  - notifications: 3.0.0
  - oauth2: 1.18.0
  - password_policy: 2.0.0
  - photos: 3.0.2
  - privacy: 2.0.0
  - provisioning_api: 1.20.0
  - related_resources: 1.5.0
  - serverinfo: 2.0.0
  - settings: 1.13.0
  - sharebymail: 1.20.0
  - sociallogin: 5.6.5
  - support: 2.0.0
  - survey_client: 2.0.0
  - systemtags: 1.20.0
  - theming: 2.5.0
  - twofactor_backupcodes: 1.19.0
  - updatenotification: 1.20.0
  - user_status: 1.10.0
  - viewer: 3.0.0
  - weather_status: 1.10.0
  - workflowengine: 2.12.0
Disabled:
  - admin_audit: 1.20.0
  - dashboard: 7.10.0 (installed 7.9.0)
  - encryption: 2.18.0
  - files_external: 1.22.0
  - firstrunwizard: 3.0.0 (installed 2.18.0)
  - geoblocker: 0.5.14 (installed 0.5.14)
  - recommendations: 3.0.0 (installed 2.1.0)
  - suspicious_login: 8.0.0
  - text: 4.1.0 (installed 3.10.1)
  - twofactor_nextcloud_notification: 4.0.0
  - twofactor_totp: 12.0.0-dev
  - user_ldap: 1.21.0
  - webhook_listeners: 1.1.0-dev
root@pan:/var/www/html#
```

### Nextcloud Signing status

```shell
no error
```

### Nextcloud Logs

```shell
no
```

### Additional info

no

### PR DESCRIPTION
**Pull request by susnux**
* Resolves https://github.com/nextcloud/server/issues/47267

## Summary

As suggested by @skjnldsv in the previous PR we should not use `owner-id` directly but just use the top level `owner` attribute (which will be set to owner-id anyways by default).
This fixes the mentioned issue, moreover I added some tests for the inline status action and a regression test for this bug.

(We nevertheless should fix the wrong type of the `owner-id` attribute in `davResultToNode` and cast it to string, but we can do so later in `@nextcloud/files`)

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by Pytal — APPROVED** [review]

File: apps/files_sharing/src/actions/sharingStatusAction.ts [hunk 1]
```diff
@@ -69,7 +67,7 @@ export const action = new FileAction({
 		}
 
 		// Group shares
-		if (shareTypes.includes(ShareType.Grup)
+		if (shareTypes.includes(ShareType.Group)
```
Comment (skjnldsv) [inline apps/files_sharing/src/actions/sharingStatusAction.ts:70]: That's a good one 😅🙈

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
``` 
files_sharing: Files inline status action
    (Attempt 1 of 3) No "shared" tag when user ID is purely numerical
    (Attempt 2 of 3) No "shared" tag when user ID is purely numerical
    1) No "shared" tag when user ID is purely numerical
    ```

**Comment by susnux** [comment]
@skjnldsv 

> files_sharing: Files inline status action

Happens because Cypress does not compile the sources before testing -> So it fails like it does on master. As soon as we compile it will work here. (No idea why the compile step was removed from cypress?)

**Pull request merged**